tests/common/devices/onyx.py

```diff
@@ -196,13 +208,85 @@ def get_speed(self, interface_name):
         if 'failed' in show_int_result and show_int_result['failed']:
             logger.error('Failed to get speed for port {} - {}'.format(interface_name, show_int_result['msg']))
             return False
-        
+
         out = show_int_result['stdout'][0].strip()
         logger.debug('Get speed for port {} from onyx: {}'.format(interface_name, out))
         if not out:
             return None
-        
+
         # The output should be something like: "Actual speed:50G"
         speed = out.split(':')[-1].strip()
         pos = speed.find('G')
         return speed[:pos] + '000'
+
+    def prepare_drop_counter_config(self, match_mac, set_mac, eth_field):
+        """Set configuration for drop_packets tests if fanout has onyx OS
+        Affected tests:test_equal_smac_dmac_drop, test_multicast_smac_drop
+
+        Args:
+            match_mac (str): mac address to match in openflow rule
+            set_mac (str): mac address to which match mac should be changed
+            eth_field (str): place in which replace match mac to set_mac, usually 'eth_src'
+
+        Returns:
+            boolean: True if success. Usually, the method return False only if the operation
+            is not supported or failed.
+        """
+        trunk_port = self._get_trunk_port_to_server()
+        openflow_port_id = self._get_openflow_port_id(trunk_port)
+        cmd = 'openflow add-flows {rule_id} table=0,priority=10,dl_src={match_mac},' \
+              'in_port={openflow_port_id},actions=set_field:{set_mac}->{eth_field}'
+        out = self.host.onyx_config(lines=[cmd.format(
+            rule_id=MAX_OPENFLOW_RULE_ID, match_mac=match_mac,
+            openflow_port_id=openflow_port_id, set_mac=set_mac, eth_field=eth_field)])
+        if 'failed' in out and out['failed']:
+            logger.error('Failed to set openflow rule - {}'.format(out['msg']))
+            return False
+        logger.debug('Setting openflow rule succed from onyx: {}'.format(out))
+        return True
+
+    def get_fanout_facts(self):
+        result = self.localhost.conn_graph_facts(host=self.hostname,
+                                                 filepath=LAB_CONNECTION_GRAPH_DIR)
```

Comment:
filepath is not the preferred way to choose right graph file. in conn_graph_facts.py, there is already automation to choose a graph file to best suite the device(s) passed in. Is that method failing for you?

Reply:
Do you know why? lab file is the first one checked in the list: https://github.com/Azure/sonic-mgmt/blob/master/ansible/files/graph_files.yml#L3

Can you make the existing mechanism working.

By merging this PR, nightly test depends on the existing automation will fail.

Reply:
@yxieca I did changed the implementation of get_fanout_facts() to fixture fanout_graph_facts